ICAFIX/hcp_fix_multi_run

```diff
@@ -419,427 +430,458 @@ log_Msg "PWD: $DIR"
-		"${matlab_interpreter[@]}" <<<"$matlab_code"
-		echo
-		;;
-esac
-
-Dim=$(cat ${concatfmrihp}_dims.txt)
-
-log_Msg "Dim: ${Dim}"
-
-if [[ $ICAmethod == MELODIC ]]; then
-	## ---------------------------------------------------------------------------
-	## Run melodic on concatenated file
-	## ---------------------------------------------------------------------------
-
-	log_Debug_Msg "About to run melodic: Contents of ${concatfmrihp}.ica follow"
-	if [ ! -z "${log_debugOn}" ] ; then
-		ls -lRa ${concatfmrihp}.ica
-	fi
-
-	#grab melodic from $PATH by default, don't hardcode it with respect to $FSLDIR
-	#we need to do "if which ..." because the script currently uses ERR trap
-	if which melodic &> /dev/null
-	then
-		MELODIC=$(which melodic 2> /dev/null)
-	else
-		#if it isn't even in $PATH, fall back on FSLDIR
-		MELODIC="${FSLDIR}/bin/melodic"
-	fi
-
-	log_Msg "Running MELODIC located at: $MELODIC"
-	log_Debug_Msg "Beginning of melodic version log, help, and checksum"
-	if [ ! -z "${log_debugOn}" ] ; then
-		log_Debug_Msg "$MELODIC --version"
-		$MELODIC --version
-		log_Debug_Msg "$MELODIC --help"
-		$MELODIC --help
-		log_Debug_Msg "md5sum $MELODIC"
-		md5sum $MELODIC
-	fi
-	log_Debug_Msg "End of melodic version log, help, and checksum"
-
-	# Use the Wishart-based dim estimate instead of the melodic-based estimate (--dim=${Dim})
-	# Also turn off melodic VN and feed in the input that has already been VN'ed instead (--vn flag, which turns OFF melodic's vn)
-	# Added "-m ${concatfmri}_brain_mask" to avoid memory issue - Takuya Hayahsi July 2018
-	melodic_cmd=("${MELODIC}" -i "${concatfmrihp}_vnts" -o "${concatfmrihp}.ica/filtered_func_data.ica" --nobet --report --Oall --tr="${tr}" --vn --dim="${Dim}" -m "${concatfmri}_brain_mask")
-
-	if [ ! -z "${log_debugOn}" ] ; then
-		melodic_cmd+=(--verbose --debug)
-	fi
-
-	#keep the existing handling code for melodic errors?
-	debug_disable_trap
-
-	log_Msg "melodic_cmd: ${melodic_cmd[*]}"
-	"${melodic_cmd[@]}"
-	return_code=$?
-	log_Msg "melodic has been run: return_code = ${return_code}"
-	log_Debug_Msg "melodic has been run: Contents of ${concatfmrihp}.ica follow"
-	if [ ! -z "${log_debugOn}" ] ; then
-		ls -lRa ${concatfmrihp}.ica
-	fi
-
-	if [ "${return_code}" -ne "0" ] ; then
-		log_Err_Abort "melodic has returned a non-zero code"
+else
+    log_Msg "Skipping ICA and proceeding directly to FIX. All the files generated by ICA have to be present!"
+
+	# Set the variables
+	ConcatFolder=`dirname ${ConcatName}`
+	ConcatNameNoExt=$($FSLDIR/bin/remove_ext $ConcatName)  # No extension, but still includes the directory path
+	concatfmri=`basename ${ConcatNameNoExt}`  # Directory path is now removed
+	concatfmrihp=${concatfmri}_hp${hp}
+
+	# Switch to concat folder
+	cd ${ConcatFolder}
+
+    if [ -d "${concatfmrihp}.ica/fix" ] && [ -n "$FIXBackup" ]; then
+        pushd "${concatfmrihp}.ica"
+        mv fix $FIXBackup
+        popd
+		log_Msg "Previous fix solution found and moved to ${$FIXBackup}."
 	fi
```

Comment:
This has quite a mix of tabs and spaces.

Reply:
I have changed everything to spaces in the latest commit.